plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.test.ts

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import os from 'os';
+import { resolve as resolvePath } from 'path';
+import fs from 'fs-extra';
+import mockFs from 'mock-fs';
+import { getVoidLogger, UrlReader } from '@backstage/backend-common';
+import { ScmIntegrations } from '@backstage/integration';
+import { PassThrough } from 'stream';
+import { fetchContents } from './helpers';
+import { ActionContext, TemplateAction } from '../../types';
+import { createFetchTemplateAction, FetchTemplateInput } from './template';
+
+jest.mock('./helpers', () => ({
+  fetchContents: jest.fn(),
+}));
+
+const aBinaryFile = fs.readFileSync(
+  resolvePath(
+    'src',
+    '../fixtures/test-nested-template/public/react-logo192.png',
+  ),
+);
+
+const mockFetchContents = fetchContents as jest.MockedFunction<
+  typeof fetchContents
+>;
+
+describe('fetch:template', () => {
+  let action: TemplateAction<any>;
+
+  const workspacePath = os.tmpdir();
+  const createTemporaryDirectory: jest.MockedFunction<
+    ActionContext<FetchTemplateInput>['createTemporaryDirectory']
+  > = jest.fn(() =>
+    Promise.resolve(
+      `${workspacePath}/${createTemporaryDirectory.mock.calls.length}`,
```

Comment:
Same here, might need to use `path.join` here to get the separator for windows.

Reply:
Thanks Ben! Fixed in 886844aa3

Reply:
That doesn't look right though! Expected to see two arguments, something like

```ts
joinPath(workspacePath, String(createTemporaryDirectory.mock.calls.length))
```

Reply:
Haha - yes, just spotted the same thing! Since path.join normalizes the path after joining, I think the change in 886844a should work, but I agree it totally doesn't look right and isn't what I intended - bit hasty! Thanks for pointing it out, fixed in 51513c642.